nancylee713/AILanguageTutor_BE
# Models

## What's this PR do?
Modify `models.py`, add additional arguments (i.e., `nullable`, `unique`) and relationship between tables

### Where should the reviewer start?
Review `models.py`

### How should this be manually tested?
1. Make sure that you are working in a virtual environment (check for (venv)).
2. Run the python interpreter, by typing `python3` in the terminal.
3. Type the following inside the interpreter:

```python
# Import files and create a db
>>> from app import db
>>> from app import User, UserProfile
>>> db.create_all()

# Add a user to User model
>>> user_1 = User(email='test@email.com', password='password')
>>> db.session.add(user_1)
>>> db.session.commit()
>>> user_1
User(id: '5', email: 'test@email.com')


# Add a user_profile to UserProfile model
>>> user_1_profile = UserProfile(name='nancy', age=28, proficiency='beginner', user_id=5)
>>> db.session.add(user_1_profile)
>>> db.session.commit()


# Check relationship between user and user_profile table
>>> user_1.profile
[UserProfile(name: 'nancy', level: 'beginner', user_id: '5')]

>>> user_1_profile.user
User(id: '5', email: 'test@email.com')
```

### What are the relevant tickets?
#14 

#### Screenshots (if appropriate)


- Note: other tables that were created in the previous PR should be modified similarly.
